src/Kiota.Builder/Refiners/CSharpReservedTypesProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace Kiota.Builder.Refiners {
+    public class CSharpReservedTypesProvider : IReservedNamesProvider
+    {
+        private readonly Lazy<HashSet<string>> _reservedNames = new(() => new(StringComparer.OrdinalIgnoreCase)
+        {
+            "task",
```

Comment:
shouldn't task be removed from the reserved names one?

Reply:
Thanks. Resolved